### PR TITLE
feat(handlers/webhook): test slash command name against ^[\w-]{1,32}$

### LINKED ISSUE #567: Slash Command Name Regex
The `name` of a slash command should be checked using the regex that is documented.

https://github.com/discord/discord-api-docs/commit/668a990efe8b1e22c2cbd29ef73c2ac7d95f7234

NEW: https://github.com/discord/discord-api-docs/commit/df59cef52e162c4d225c81c76e7e022206027cfc

3-32 character name matching `^(?:\w|[-_])+$`

### PR DESCRIPTION
**Pull request by ayntee**
Closes #567

### CODE REVIEW

File: src/api/handlers/webhook.ts [hunk 1]
```diff
@@ -281,8 +281,7 @@ export async function deleteWebhookMessage(
  * Guild commands update **instantly**. We recommend you use guild commands for quick testing, and global commands when they're ready for public use.
  */
 export async function createSlashCommand(options: CreateSlashCommandOptions) {
-  // Use ... for content length due to unicode characters and js .length handling
-  if ([...options.name].length < 2 || [...options.name].length > 32) {
+  if (/^[\w-]{1,32}$/.test(options.name)) {
```
Comment (itohatweb) [inline src/api/handlers/webhook.ts:284]: We need to check if the name does not match the regex.
```suggestion
  if (!/^[\w-]{1,32}$/.test(options.name)) {
```

File: src/api/handlers/webhook.ts [hunk 2]
```diff
@@ -335,8 +334,7 @@ export async function upsertSlashCommand(
   options: UpsertSlashCommandOptions,
   guildID?: string,
 ) {
-  // Use ... for content length due to unicode characters and js .length handling
-  if ([...options.name].length < 2 || [...options.name].length > 32) {
+  if (/^[\w-]{1,32}$/.test(options.name)) {
```
Comment (itohatweb) [inline src/api/handlers/webhook.ts:337]: We need to check if the name does not match the regex.
```suggestion
  if (!/^[\w-]{1,32}$/.test(options.name)) {
```

File: src/api/handlers/webhook.ts [hunk 3]
```diff
@@ -370,8 +368,7 @@ export async function upsertSlashCommands(
   guildID?: string,
 ) {
   const data = options.map((option) => {
-    // Use ... for content length due to unicode characters and js .length handling
-    if ([...option.name].length < 2 || [...option.name].length > 32) {
+    if (/^[\w-]{1,32}$/.test(option.name)) {
```
Comment (itohatweb) [inline src/api/handlers/webhook.ts:371]: We need to check if the name does not match the regex.
```suggestion
    if (!/^[\w-]{1,32}$/.test(option.name)) {
```

File: src/api/handlers/webhook.ts [hunk 4]
```diff
@@ -404,8 +401,7 @@ export async function editSlashCommand(
   options: EditSlashCommandOptions,
   guildID?: string,
 ) {
-  // Use ... for content length due to unicode characters and js .length handling
-  if ([...options.name].length < 2 || [...options.name].length > 32) {
+  if (/^[\w-]{1,32}$/.test(options.name)) {
```
Comment (itohatweb) [inline src/api/handlers/webhook.ts:404]: We need to check if the name does not match the regex.
```suggestion
  if (!/^[\w-]{1,32}$/.test(options.name)) {
```

**Review by itohatweb — CHANGES_REQUESTED** [review]

**Review by itohatweb — DISMISSED** [review]
LGTM

File: src/api/handlers/webhook.ts [hunk 5]
```diff
@@ -281,8 +281,7 @@ export async function deleteWebhookMessage(
  * Guild commands update **instantly**. We recommend you use guild commands for quick testing, and global commands when they're ready for public use.
  */
 export async function createSlashCommand(options: CreateSlashCommandOptions) {
-  // Use ... for content length due to unicode characters and js .length handling
-  if ([...options.name].length < 2 || [...options.name].length > 32) {
+  if (!/^[\w-]{1,32}$/.test(options.name)) {
```
Comment (Skillz4Killz) [inline src/api/handlers/webhook.ts:284]: Can we move this regex into the constants file?

Comment (ayntee) [reply src/api/handlers/webhook.ts:284]: Sure, will do!

**Review by itohatweb — APPROVED** [review]

**Review by Skillz4Killz — APPROVED** [review]

**Pull request merged**